action/sealedenvelope.go

```diff
@@ -51,10 +77,21 @@ func (sealed *SealedEnvelope) LoadProto(pbAct *iotextypes.Action) error {
 		return errors.New("nil action to load proto")
 	}
 
+	encoding := pbAct.GetEncoding()
+	if !IsValidEncoding(encoding) {
+		return errors.Errorf("invalid encoding = %v", encoding)
+	}
 	var elp Envelope = &envelope{}
 	if err := elp.LoadProto(pbAct.GetCore()); err != nil {
 		return err
 	}
+	if IsRLP(encoding) {
+		// verify action type can support RLP-encoding
+		if _, err := actionToRLP(elp.Action()); err != nil {
+			return err
+		}
+		sealed.externChainID = config.ExternChainID()
+	}
```

Comment:
if we need to do validation for rlp encoding, should we do it for protobuf as well?

Reply:
for protobuf, validation is done in Envelope.Load(), it will check all possible types.

here is to make sure for RLP case, the struct{} inside must be Transfer or Execution, i.e., to catch someone send in a StakeCreate with encoding = 1

Reply:
ok